crates/rspack_plugin_runtime/src/runtime_plugin.rs

```diff
@@ -19,6 +21,99 @@ use crate::runtime_module::{
   PublicPathRuntimeModule, SystemContextRuntimeModule,
 };
 
+static GLOBALS_ON_REQUIRE: Lazy<Vec<RuntimeGlobals>> = Lazy::new(|| {
+  vec![
+    // RuntimeGlobals::CHUNK_NAME,
+    // RuntimeGlobals::RUNTIME_ID,
+    RuntimeGlobals::COMPAT_GET_DEFAULT_EXPORT,
+    RuntimeGlobals::CREATE_FAKE_NAMESPACE_OBJECT,
+    RuntimeGlobals::CREATE_SCRIPT,
+    RuntimeGlobals::CREATE_SCRIPT_URL,
+    RuntimeGlobals::GET_TRUSTED_TYPES_POLICY,
+    RuntimeGlobals::DEFINE_PROPERTY_GETTERS,
+    RuntimeGlobals::ENSURE_CHUNK,
+    RuntimeGlobals::ENTRY_MODULE_ID,
+    RuntimeGlobals::GET_FULL_HASH,
+    RuntimeGlobals::GLOBAL,
+    RuntimeGlobals::MAKE_NAMESPACE_OBJECT,
+    RuntimeGlobals::MODULE_CACHE,
+    RuntimeGlobals::MODULE_FACTORIES,
+    RuntimeGlobals::MODULE_FACTORIES_ADD_ONLY,
+    RuntimeGlobals::INTERCEPT_MODULE_EXECUTION,
+    RuntimeGlobals::PUBLIC_PATH,
+    RuntimeGlobals::BASE_URI,
+    // RuntimeGlobals::RELATIVE_URL,
+    // RuntimeGlobals::SCRIPT_NONCE,
+    // RuntimeGlobals::UNCAUGHT_ERROR_HANDLER,
+    RuntimeGlobals::ASYNC_MODULE,
+    // RuntimeGlobals::WASM_INSTANCES,
+    RuntimeGlobals::INSTANTIATE_WASM,
+    // RuntimeGlobals::SHARE_SCOPE_MAP,
+    // RuntimeGlobals::INITIALIZE_SHARING,
+    RuntimeGlobals::LOAD_SCRIPT,
+    // RuntimeGlobals::SYSTEM_CONTEXT,
+    RuntimeGlobals::ON_CHUNKS_LOADED,
+  ]
+});
+
+static MODULE_DEPENDENCIES: Lazy<Vec<(RuntimeGlobals, Vec<RuntimeGlobals>)>> = Lazy::new(|| {
+  vec![
+    (RuntimeGlobals::MODULE_LOADED, vec![RuntimeGlobals::MODULE]),
+    (RuntimeGlobals::MODULE_ID, vec![RuntimeGlobals::MODULE]),
+  ]
+});
+
+static TREE_DEPENDENCIES: Lazy<Vec<(RuntimeGlobals, Vec<RuntimeGlobals>)>> = Lazy::new(|| {
```

Comment:
Can not use hashmap because the sequence of iter() is not stable

Reply:
@LingyuCoder  Maybe `LinkedHashMap`?